ircd/client.c

```diff
@@ -420,6 +420,24 @@ check_pings_list(rb_dlink_list * list)
 				client_p->localClient->lasttime = rb_current_time() - ping;
 				sendto_one(client_p, "PING :%s", me.name);
 			}
+			else if (!(client_p->flags & FLAGS_PINGWARN) && ConfigFileEntry.ping_warn_time > 0 &&
```

Comment:
These warnings should repeat until the split or recovery.  This is what oftc-hybrid does, and it eliminates uncertainty about link status (and makes them much less likely to miss in the noise).  Granted, oftc-hybrid runs this every 5 seconds and solanum runs it every 30 seconds, but that's still better than nothing.

Reply:
i'd prefer it warn once when it passes the threshold and then also snote when it's back. you can infer it's still dying that way

Reply:
Only if you actually see the snote. I have a 10 line split window for snotes; the average life of what's visible is anywhere from 15 to 60 seconds. I'd rather have an snote repeated every 30 seconds when a server's not responsive and have a better chance of seeing it than to have an snote that only fires once and then disappears off my screen in 15 seconds never to be seen again until the split happens.